### PR TITLE
Added command line support for using local CEF builds

### PR DESCRIPTION
**Pull request by spazzarama**
Example for vs2015 and nupkg-only with local custom build of Cef branch 2924
```
.\build.ps1 -Target vs2015 -DownloadBinary local -CefVersion 3.2924.1538.gbfdeccd -CefBinaryDir "../../cefsource/chromium/src/cef/binary_distrib/"
.\build.ps1 -Target nupkg-only -DownloadBinary none -CefVersion 3.2924.1538.gbfdeccd
```

### CODE REVIEW

**Comment by amaitland** [comment]
- Formatting is a little out, spaces not tabs (Review using the `Files Changed` tab above).
- Braces on new lines.

Apart from that looks fine, usage examples are a nice touch 👍

**Comment by spazzarama** [comment]
@amaitland ah yep sorry, edited in the crappy Windows Powershell ISE

**Comment by spazzarama** [comment]
@amaitland:
On review - there are no tabs, it is inconsistent spacing that was already there (i.e. it is consistently inconsistent inline with the existing Download function that it is a copy of). Unless I am missing which one you are referring to.

I'm assuming you are referring to the braces in the switch statement (I've changed now to reflect other switch statements. But again it is fairly inconsistent through out.

I'll await further feedback before committing. I'm not going to change any existing functions (which have 2 spaces instead of 4, inconsistent braces), but if you would like me to adjust what I have to a particular style then please advice correct style (my guess, 4 spaces indent, braces begin on new line).

File: build.ps1 [hunk 1]
```diff
@@ -461,9 +542,18 @@ function CheckDependencies()
 
 CheckDependencies
 
-if($DownloadBinary)
+switch -Exact ($DownloadBinary)
 {
-  DownloadCefBinaryAndUnzip
+    "none" {
```
Comment (amaitland) [inline build.ps1]: Brace on new line

File: build.ps1 [hunk 2]
```diff
@@ -444,6 +451,80 @@ function DownloadCefBinaryAndUnzip()
...
+	# Extract bzip file
+	sz e $LocalFile
+	# Extract tar file
+	$TarFile = ($LocalFile).Substring(0, $LocalFile.length - 4)
+	sz x $TarFile
+	#Sleep for a short period to allow 7z to release it's file handles
+	sleep -m 2000
+	#Remove tar file
+	Remove-Item $TarFile
+    $Folder = Join-Path $WorkingDir ($Cef64FileName.Substring(0, $Cef64FileName.length - 8))
```
Comment (amaitland) [inline build.ps1]: Indentation is out. Assuming you copied this and the original indentation is out and you have time please fix that also.

File: build.ps1 [hunk 3]
```diff
@@ -444,6 +451,80 @@ function DownloadCefBinaryAndUnzip()
...
+    Remove-Item $Folder
+  }
+  
+  $LocalFile = Join-Path $WorkingDir $Cef64FileName
+  
+  if(-not (Test-Path $LocalFile))
+  {
+	
+	Write-Diagnostic "Copy $Cef64FileName (approx 200mb)"
+    Copy-Item ($CefBuildDir+$Cef64FileName) $LocalFile;
```
Comment (amaitland) [inline build.ps1]: Indentation

File: build.ps1 [hunk 4]
```diff
@@ -444,6 +451,80 @@ function DownloadCefBinaryAndUnzip()
...
+	# Extract bzip file
+   	sz e $LocalFile
+	# Extract tar file
+	$TarFile = ($LocalFile).Substring(0, $LocalFile.length - 4)
+	sz x $TarFile
+	#Sleep for a short period to allow 7z to release it's file handles
+	sleep -m 2000
+	#Remove tar file
+	Remove-Item $TarFile
+    $Folder = Join-Path $WorkingDir ($Cef32FileName.Substring(0, $Cef32FileName.length - 8))
```
Comment (amaitland) [inline build.ps1]: Indentation

**Comment by amaitland** [comment]
Merging now, I'll worry about the formatting later 👍

**Comment by spazzarama** [comment]
Ok no worries, nearly finished, but yeah might be better to do the whole file in one go. Enjoy!

**Comment by amaitland** [comment]
> but yeah might be better to do the whole file in one go

Yep, it's a mess! Oh well. A problem for another day.

**Pull request merged**